### PR TITLE
Breadcrumb refactor and touchups

### PR DESCRIPTION
**Pull request by shhsu**

### CODE REVIEW

**Comment by shhsu** [comment]
This PR create new breadcrumb, fixes performance issue and fix some compiler warning and compile time analysis issue

File: src/app/common/http-client.ts [hunk 1]
```diff
@@ -80,11 +80,12 @@ export class HttpClient {
...
-            take(1),
-            catchError(e => {
-                if (this.runtime.HandleConnectError(e)) {
-                    return throwError(e);
+            catchError((e, _) => {
+                var unhandled = this.runtime.HandleConnectError(e);
+                if (unhandled) {
+                    return throwError(unhandled);
                 }
+                return of(null);
```
Comment (jhkimnew) [inline src/app/common/http-client.ts:89]: please add some comment why we ignore exception error here.

File: src/app/diagnostics/logger.ts [hunk 2]
```diff
@@ -44,16 +45,34 @@ class ConsoleLogger implements Logger {
     }
 }
 
+export function instrument<T>(logger: Logger, name: string): OperatorFunction<T, T> {
```
Comment (jhkimnew) [inline src/app/diagnostics/logger.ts:49]: please add some comment how we can use this utility function in debugging time and we don't use actually at this point in time.

File: src/app/header/breadcrumbs.component.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    public Crumbs: Breadcrumb[] = [];
+    private destroy: Subject<boolean> = new Subject<boolean>();
+
+    constructor(
+        private router: Router,
+        private factory: LoggerFactory,
+        private srv: BreadcrumbsService,
+    ){
+        let logger = factory.Create(this);
+        srv.crumbs.pipe(
```
Comment (jhkimnew) [inline src/app/header/breadcrumbs.component.ts:56]: add some comment about the difference with other coding style

File: src/app/main/app.component.ts [hunk 4]
```diff
@@ -76,14 +63,20 @@ export class AppComponent implements OnInit {
         return this._router.isActive(route, true);
     }
 
+    onActivate(component) {
+        if (!Object.getOwnPropertyNames(component).find(name => component[name] instanceof BreadcrumbsService)) {
```
Comment (jhkimnew) [inline src/app/main/app.component.ts:69]: add some comment for the purpose of this function

**Comment by jhkimnew** [comment]
Accessibility test for breadcrumb required.

File: src/app/resources/scripts/iis_scripts/Initialize-AdminAPI.ps1 [hunk 5]
```diff
@@ -237,9 +242,9 @@ LogVerbose 'Started admin_api_util...'
 if ($install) {
     InstallComponents
 } else {
-    $app = GetIISAdminAPIApp
-    if ($app) {
-        EnsureMinVersion $app $appMinVersion
+    $appInfo = Get-Item "HKLM:\SYSTEM\CurrentControlSet\Services\$serviceName" -ErrorAction SilentlyContinue
```
Comment (jhkimnew) [inline src/app/resources/scripts/iis_scripts/Initialize-AdminAPI.ps1:245]: Please check if Downlevel OS is using the same registry key path.

File: src/app/runtime/wac/services/powershell-service.ts [hunk 6]
```diff
@@ -1,12 +1,13 @@
 import { Injectable, Inject } from '@angular/core'
 import { AppContextService } from '@microsoft/windows-admin-center-sdk/angular'
-import { PowerShell, PowerShellSession } from '@microsoft/windows-admin-center-sdk/core'
+import { PowerShell, PowerShellSession, PowerShellResult } from '@microsoft/windows-admin-center-sdk/core'
 import { Observable } from 'rxjs'
 import { PowerShellScripts } from 'generated/powershell-scripts'
 import { Request, Response, ResponseOptions, Headers } from '@angular/http'
 import { WACInfo } from 'runtime/runtime.wac'
-import { LoggerFactory, Logger, LogLevel, logError } from 'diagnostics/logger'
+import { LoggerFactory, Logger, LogLevel, logError, instrument } from 'diagnostics/logger'
```
Comment (jhkimnew) [inline src/app/runtime/wac/services/powershell-service.ts:8]: clean up instrument if not used.

File: src/app/webserver/app-pools/app-pool.component.ts [hunk 7]
```diff
@@ -16,10 +18,6 @@ import {AppPoolsService} from './app-pools.service';
         <app-pool-header *ngIf="pool" [pool]="pool" class="crumb-content" [class.sidebar-nav-content]="_options.active"></app-pool-header>
 
         <div *ngIf="pool" class="sidebar crumb" [class.nav]="_options.active">
-            <ul class="crumbs sme-focus-zone">
-                <li><a [routerLink]="['/webserver']">Web Server</a></li>
-                <li><a [routerLink]="['/webserver/application-pools']">Application Pools</a></li>
-            </ul>
             <vtabs [markLocation]="true" (activate)="_options.refresh()">
                 <item [name]="'General'" [ico]="'fa fa-wrench'">
                     <app-pool-general [pool]="pool" (modelChanged)="onModelChanged()"></app-pool-general>
```
Comment (jhkimnew) [inline src/app/webserver/app-pools/app-pool.component.ts:23]: sme-focus-zone is required for accessibility. Please add this to a proper location.

**Review by jhkimnew — APPROVED** [review]

File: src/app/main/app.component.ts (hunk 4 above)
Comment (shhsu) [reply src/app/main/app.component.ts:69]: added

**Pull request merged**